### ISSUE TITLE
feature request: allowe `use_github()` to create "internal" repositories

### ISSUE THREAD

**Issue by dragosmg** [r-lib/usethis#1458]
First of all, many thanks for this awesome package. I bumped into this use case a number of times and, as far as I can tell, this feature request doesn't seem to have been discussed previously. Apologies if it was and a conclusion not to implement `visibility` had been reached.

Currently the only option when using `use_github()` is to create either a `public` or `private` repository. For enterprise accounts, there often is a 3rd use case, namely to create an `internal` repo. This is `private` as far as the rest of the world is concerned, but `public` from the point of view of the members of the organisation.  

I think it might be a case of adding an argument named `visibility` with the possible values `"public"`, `"private"` or `"internal"` (keeping in line with the [GitHub POST request parameters](https://docs.github.com/en/rest/reference/repos#create-an-organization-repository--parameters)). The default value for this argument could be `"public"` or `NULL`. Moreover, as far as I understand, the [GitHub docs](https://docs.github.com/en/rest/reference/repos) state `visibility` overrides `private` in case of conflict. What do you think?

The expected behaviour would be:

```r
# creates an "internal" repo
use_github(organisation =  "My_awesome_organisation", private = FALSE, visibility = "internal")

# creates an "internal" repo as visibility overrides "private"
use_github(organisation =  "My_awesome_organisation", private = TRUE, visibility = "internal")

# use_github should error when organisation = NULL and visibility = "internal"
use_github(visibility = "internal")
```

**Comment by dragosmg** [comment]
Happy to try and submit a PR if desired.

**Comment by jennybc** [comment]
In principle, this sounds fine (what you propose and a PR). I think this function probably predates the more complicated world we now live in with GitHub Enterprise. It definitely predates usethis having explicit support for GHE.

**Comment by jennybc** [comment]
@dragosmg I implemented this. Or at least I think I did. I do have access to a GHE instance and `use_github(organisation = "ORG", visibility = "internal")` does **not** appear to create a repo with "internal" visibility. I don't know why. I _think_ the GHE I'm on is recent enough to support this. But I have no idea if there are other configuration settings at play.

Will you try this out and see if it works for you? Or if you can figure out what's wrong?

**Comment by jennybc** [comment]
Helpful links:

https://developer.github.com/changes/2019-12-03-internal-visibility-changes/

Among other interesting points, there's this:

> To access the visibility field for any of these endpoints, you must provide a custom media type in the Accept header:
>
> application/vnd.github.nebula-preview+json

And also a setting that exists for an organization:

* `members_can_create_internal_repositories`

Found this via reading some posts in the community support site: https://github.community/search?q=internal%20visibility%20API

**Comment by jennybc** [comment]
@dragosmg This now works for me on an instance of GHE version 2.22. It would be interesting to hear from your or others if this works as expected, especially if you're on a more recent version of GHE.

**Comment by jennybc** [comment]
I'm talking about this PR: https://github.com/r-lib/usethis/pull/1505

**Comment by dragosmg** [comment]
Thanks a lot for picking this up @jennybc. Unfortunately I haven't had time to focus on this. Moreover, I no longer have access to a GHE instance. I'm starting a new role tomorrow and will have to look into the tech stack at the new place.

**Comment by jennybc** [comment]
No worries. @ijlyttle is this something you care about?

**Comment by jennybc** [comment]
Yeah I am in the same boat. I can create one of these "internal" repos, because I have access to GHE through UBC. But without a genuine use case, I'm at a loss as to what to do with it. I will just wait and see if anyone uses this and finds something deficient about it or its documentation. Thanks for giving it a test drive.
